environments/dev/main.jsonnet

```diff
@@ -1,4 +1,92 @@
-local obs = (import '../base/observatorium.jsonnet');
+local dex = (import '../../components/dex.libsonnet') + {
+  config+:: {
+    local cfg = self,
+    name: 'dex',
+    namespace: 'dex',
+    config+: {
+      oauth2: {
+        passwordConnector: 'local',
+      },
+      staticClients: [
+        {
+          id: 'test',
+          name: 'test',
+          secret: 'ZXhhbXBsZS1hcHAtc2VjcmV0',
+        },
+      ],
+      enablePasswordDB: true,
+      staticPasswords: [
+        {
+          email: 'admin@example.com',
+          // bcrypt hash of the string "password"
+          hash: '$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W',
+          username: 'admin',
+          userID: '08a8684b-db88-4b73-90a9-3cd1661f5466',
+        },
+      ],
+    },
+    version: 'v2.24.0',
+    image: 'quay.io/dexidp/dex:v2.24.0',
+    commonLabels+:: {
+      'app.kubernetes.io/instance': 'e2e-test',
+    },
+  },
+};
+
+local obs = (import '../base/observatorium.jsonnet') + {
+  api+: {
+    config+: {
+      rbac: {
+        roles: [
+          {
+            name: 'read-write',
+            resources: [
+              'logs',
+              'metrics',
+            ],
+            tenants: [
+              'test',
+            ],
+            permissions: [
+              'read',
+              'write',
+            ],
+          },
+        ],
+        roleBindings: [
+          {
+            name: 'test',
+            roles: [
+              'read-write',
+            ],
+            subjects: [
+              dex.config.config.staticPasswords[0].email,
+            ],
+          },
+        ],
+      },
+      tenants: {
+        tenants: [
+          {
+            name: dex.config.config.staticClients[0].name,
+            id: '1610b0c3-c509-4592-a256-a1871353dbfa',
+            oidc: {
```

Comment:
Missing `redirectURL` here? Or omitempty in the spec?

Reply:
The only required values are the client ID and issuer URL. The secret, username claim, and redirect URL are all optional, so we can set them to omitempty.  Strictly speaking this isn't necessary since todas implementation of OIDC in observstorium API is to unmarshal these fields into strings rather than string pointers, so omitting the field is the same as setting `''` but this could obviously change

Reply:
Sure let's put them on omitempty.